### PR TITLE
Dom mutation was not happening for elements with dangerouslySetInnerHTML.

### PR DESCRIPTION
**Pull request by harish2704**
For elements with dangerouslySetInnerHTML attr, 

initial rendering will take place correctly. ( only Test case for initial render is present in the unit tests ).

after that, after some updating, if we are again rendering the same element, It was not working. 

This pull request fixes that issue. ( Unit test also included )

### CODE REVIEW

**Comment by harish2704** [comment]
[INFO] This bug can be visualized by simply running the project.
https://github.com/harish2704/preact-router5-boilerplate

If we repeat the same with react library, ( just comment resolve part in webpack config ) it will work properly

-Thanks

**Comment by developit** [comment]
Hi @harish2704! Thanks for the bug report, repro and PR (all in one)!
I need to check if this will cause excessive recycling, and if so might have to change the technique a little bit.  Regardless of that though, the tests and reproduction will help make sure any tweaks I have to make keep the functionality working :)

FWIW, I was hoping this would be fixed by the fix for #446.  I'll try to pull your tests in to see if that's the case, since that would be a super quick solution :)

File: src/vdom/diff.js [hunk 1]
```diff
@@ -315,7 +315,17 @@ function diffAttributes(dom, attrs, old) {
 	// add new & update changed attributes
 	if (attrs) {
 		for (let name in attrs) {
-			if (name!=='children' && name!=='innerHTML' && (!(name in old) || attrs[name]!==(name==='value' || name==='checked' ? dom[name] : old[name]))) {
+			if (
+				name!=='children' &&  // break if attributeName is children
+				name!=='innerHTML' &&  // break if attributeName is innerHTML
+				(
+					( name=== 'dangerouslySetInnerHTML' ) || // continue if attributeName is dangerouslySetInnerHTML
```
Comment (harish2704) [inline src/vdom/diff.js]: This line is the essence of PR #454

**Comment by harish2704** [comment]
Hi,
In a quick test, the issue didn't seems to be resolved even after the merge.

I tested using this sample code against current master branch (bc84e2a6a41e4cfd033f5497a7d7fa9d99c9e486).

-info

**Comment by developit** [comment]
@harish2704 what should I be looking for when I run the example?

**Comment by developit** [comment]
I was able to get your repo running, just not really sure what I'm looking for.

**Comment by harish2704** [comment]
Hi,
In the demo , the home route rendering using  **dangerouslySetInnerHTML**

If you try to goback to home route after visiting any other routes, You can see the problem.

It was not rendering

4preact

**Pull request merged**